### PR TITLE
intentions included

### LINKED ISSUE #53: Create intentions from service `nifi`=> `nifi-registry`
## Feature description
1) Create intentions from service-to-service: `nifi`=> `nifi-registry` (in Ansible playbook: `/dev/Ansible`)
2) Add intentions documentation in README.md

## Checklist (after created issue)
- [x] Added label(s)
- [ ] Added to project
- [x] Added to milestone

### PR DESCRIPTION
**Pull request by AditiSharma77**
## Closes/fixes/resolves issue(s)?
Closes #53 
## Checklist (after created PR)
- [x] Added reviewers
- [x] Added assignee(s)
- [x] Added label(s)
- [ ] Added to project
- [x] Added to milestone

### CODE REVIEW

File: dev/ansible/00_create_intention.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+- name: set variable
+  set_fact:
+    consul_token: "{{ lookup('env', 'consul_master_token') }}"
+
+- name: Check if intention nifi => nifi_registry (allow) exists
+  shell: consul intention match -source nifi
+  register: nifi_intention
+  when: lookup('env', 'consul_acl') | bool and lookup('env', 'consul_acl_default_policy') == 'deny'
+  environment:
+    CONSUL_HTTP_TOKEN: "{{ consul_token }}"
```
Comment (hannemariavister) [inline dev/ansible/00_create_intention.yml]: You have wrong consul_token, it must be `'consul_master_token'`. I suggest that you also include env-variable `CONSUL_HTTP_ADDR` such that:
```
environment:
    CONSUL_HTTP_ADDR: "{{ lookup('env', 'CONSUL_HTTP_ADDR') }}"
    CONSUL_HTTP_TOKEN: "{{ lookup('env', 'consul_master_token') }}"
```

File: dev/ansible/00_create_intention.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+- name: Consul create intention nifi => nifi_registry (allow)
+  shell: consul intention create nifi nifi_registry
+  when: lookup('env', 'consul_acl') | bool and lookup('env', 'consul_acl_default_policy') == 'deny' and nifi_intention.stdout_lines == []
+  environment:
+    CONSUL_HTTP_TOKEN: "{{ consul_token }}"
```
Comment (hannemariavister) [inline dev/ansible/00_create_intention.yml]: Change to:
```
environment:
    CONSUL_HTTP_ADDR: "{{ lookup('env', 'CONSUL_HTTP_ADDR') }}"
    CONSUL_HTTP_TOKEN: "{{ lookup('env', 'consul_master_token') }}"
```

File: dev/ansible/00_create_intention.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+- name: Check if intention nifi-local => nifi (allow) exists
+  shell: consul intention match -source nifi-local
+  register: nifi_local_intention
+  when: lookup('env', 'consul_acl') | bool and lookup('env', 'consul_acl_default_policy') == 'deny'
+  environment:
+    CONSUL_HTTP_TOKEN: "{{ consul_token }}"
+
```
Comment (hannemariavister) [inline dev/ansible/00_create_intention.yml]: Suggestion: 
```
environment:
    CONSUL_HTTP_ADDR: "{{ lookup('env', 'CONSUL_HTTP_ADDR') }}"
    CONSUL_HTTP_TOKEN: "{{ lookup('env', 'consul_master_token') }}"
```

File: dev/ansible/00_create_intention.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+- name: Consul create intention nifi-local => nifi (allow)
+  shell: consul intention create nifi-local nifi
+  when: lookup('env', 'consul_acl') | bool and lookup('env', 'consul_acl_default_policy') == 'deny' and nifi_local_intention.stdout_lines == []
+  environment:
+    CONSUL_HTTP_TOKEN: "{{ consul_token }}"
```
Comment (hannemariavister) [inline dev/ansible/00_create_intention.yml]: Suggestion: 
```
environment:
    CONSUL_HTTP_ADDR: "{{ lookup('env', 'CONSUL_HTTP_ADDR') }}"
    CONSUL_HTTP_TOKEN: "{{ lookup('env', 'consul_master_token') }}"
```

File: dev/ansible/00_create_intention.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+- name: Check if intention nifi => nifi_registry (allow) exists
+  shell: consul intention match -source nifi
+  register: nifi_intention
+  when: lookup('env', 'consul_acl') | bool and lookup('env', 'consul_acl_default_policy') == 'deny'
+  environment:
+    CONSUL_HTTP_TOKEN: "{{ consul_token }}"
+
+- name: Consul create intention nifi => nifi_registry (allow)
+  shell: consul intention create nifi nifi_registry
```
Comment (hannemariavister) [inline dev/ansible/00_create_intention.yml]: I think you have wrong service name for nifi-registry? Shouldn’t it be `nifi-registry` (not `nifi_registry`)?

**Review by hannemariavister — CHANGES_REQUESTED** [review]

**Review by hannemariavister — COMMENTED** [review]
It seems that the intentions are not created. See my comments above.

File: dev/ansible/00_create_intention.yml (hunk 1 above)
Comment (AditiSharma77) [reply dev/ansible/00_create_intention.yml]: I have included consul_http_addr , but if you check line 3 consul_token: "{{ lookup('env', 'consul_master_token') }}", then this line number 3 works as same   for  line 10 .
CONSUL_HTTP_TOKEN: "{{ consul_token }}" means  consul_token: "{{ lookup('env', 'consul_master_token') }}"

File: dev/ansible/00_create_intention.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
+- name: set variable
+  set_fact:
+    mode: "{{ mode }}"
```
Comment (hannemariavister) [inline dev/ansible/00_create_intention.yml]: Is it necessary to create this variable `mode` ? I think it is unused?

**Review by hannemariavister — CHANGES_REQUESTED** [review]

File: dev/ansible/00_create_intention.yml (hunk 6 above)
Comment (karimacherradi) [reply dev/ansible/00_create_intention.yml]: This is new! I agree with HM

**Review by karimacherradi — APPROVED** [review]
Good :100: :+1:

**Review by hannemariavister — APPROVED** [review]
LGTM! :-)

**Pull request merged**